api/src/routes/users/@me/channels.ts

```diff
@@ -1,15 +1,21 @@
-import { Router, Request, Response } from "express";
-import { Channel, ChannelCreateEvent, ChannelType, Snowflake, trimSpecial, User, emitEvent, Recipient } from "@fosscord/util";
-import { HTTPError } from "lambert-server";
+import { Request, Response, Router } from "express";
+import { PublicUserProjection, Recipient, User, ChannelService } from "@fosscord/util";
 import { route } from "@fosscord/api";
-import { In } from "typeorm";
 
 const router: Router = Router();
 
 router.get("/", route({}), async (req: Request, res: Response) => {
-	const recipients = await Recipient.find({ where: { user_id: req.user_id }, relations: ["channel"] });
+	const recipients = await Recipient.find({ where: { user_id: req.user_id }, relations: ["channel", "user"] });
 
-	res.json(recipients.map((x) => x.channel));
+	//TODO check if this is right
+	const aa = await Promise.all(recipients.map(async (x) => {
+		return {
+			...(x.channel),
+			recipients: await User.findOneOrFail({ where: { id: x.user_id }, select: PublicUserProjection }),
```

Comment:
You could use the user relation of recipients instead, so that you don't need to fetch them again.
Tho if you do this, you need to add the PublicUserProjection to the selection the the Recipient.find query.

Reply:
I changed it to use the DmChannelDTO, some Recipients will be fetched twice (or maybe typeorm has some kind of optimization in this regard) but i don't think that it matters that much performance wise since i'm not even sure that the Discord client call this API